src/main/scala/com.mediative.eigenflow/EigenflowBootstrap.scala

```diff
@@ -61,17 +61,22 @@ trait EigenflowBootstrap {
   private def stopSystem(system: ActorSystem)(code: Int): Unit = {
     Try(messagingSystem.stop())
 
-    // in case if termination process hangs
-    try {
-      val terminationProcess = system.terminate()
-
-      Await.result(terminationProcess, 10.seconds)
-      System.exit(code)
-    } catch {
-      case t: Throwable =>
-        t.printStackTrace()
-        System.err.println("Timeout actor system termination. Forcing system exit.")
-        Runtime.getRuntime.halt(1) // force quit with status code 1
+    val terminationProcess = system.terminate()
+
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    val timeoutFuture = Future { //in case system.terminate hangs
+      Thread.sleep(10000)
+      throw new TimeoutException("actorsystem.terminate timed out.  Force shutdown.")
     }
+
+    Future.firstCompletedOf(List(terminationProcess, timeoutFuture)).onComplete {
+      case Success(_) =>
+        System.exit(code)
+      case Failure(ex) =>
+        ex.printStackTrace()
+        System.exit(1)
```

Comment:
this still may hang the process, that's why `Runtime.getRuntime.halt(1)` was used